presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloClient.java

```diff
@@ -0,0 +1,1160 @@
+
+        // Creates and sets CONNECTOR in the event it has not yet been created
+        this.auths = getAccumuloConnector(this.conf).securityOperations().getUserAuthorizations(conf.getUsername());
+
+        // Create the index lookup utility
+        this.sIndexLookup = new IndexLookup(conn, conf, this.auths);
+    }
+
+    /**
+     * Creates a new AccumuloTable based on the given metadata.
+     *
+     * @param meta Metadata for the table
+     * @return {@link AccumuloTable} for the newly created table
+     */
+    public AccumuloTable createTable(ConnectorTableMetadata meta)
+    {
+        // Validate the DDL is something we can handle
+        validateCreateTable(meta);
+
+        Map<String, Object> tableProperties = meta.getProperties();
+        String rowIdColumn = getRowIdColumn(meta);
+
+        // Get the list of column handles
+        List<AccumuloColumnHandle> columns = getColumnHandles(meta, rowIdColumn);
+
+        // Create the AccumuloTable object
+        AccumuloTable table = new AccumuloTable(meta.getTable().getSchemaName(),
+                meta.getTable().getTableName(), columns, rowIdColumn,
+                AccumuloTableProperties.isExternal(tableProperties),
+                AccumuloTableProperties.getSerializerClass(tableProperties),
+                AccumuloTableProperties.getScanAuthorizations(tableProperties));
+
+        // First, create the metadata
+        metaManager.createTableMetadata(table);
+
+        // Make sure the namespace exists
+        tableManager.ensureNamespace(table.getSchema());
+
+        // Create the Accumulo table if it does not exist (for 'external' table)
+        if (!tableManager.exists(table.getFullTableName())) {
+            tableManager.createAccumuloTable(table.getFullTableName());
+        }
+
+        // Set any locality groups on the data table
+        setLocalityGroups(tableProperties, table);
+
+        // Create index tables, if appropriate
+        createIndexTables(table);
+
+        return table;
+    }
+
+    /**
+     * Validates the given metadata for a series of conditions to ensure the table is well-formed
+     *
+     * @param meta Table metadata
+     */
+    private void validateCreateTable(ConnectorTableMetadata meta)
+    {
+        validateColumns(meta);
+        validateLocalityGroups(meta);
+        if (!AccumuloTableProperties.isExternal(meta.getProperties())) {
+            validateInternalTable(meta);
+        }
+    }
+
+    /**
+     * Validates the columns set in the configure locality groups (if any) exist
+     *
+     * @param meta Table metadata
+     */
+    private void validateColumns(ConnectorTableMetadata meta)
+    {
+        // Here, we make sure the user has specified at least one non-row ID column
+        // Accumulo requires a column family and qualifier against a row ID, and these values
+        // are specified by the other columns in the table
+        if (meta.getColumns().size() == 1) {
```

Comment:
This code is only called when a new table is created; not sure when zero columns would be given.  But in any case it can certainly be `<= 1`.  I'll make the change.